fxa-client/sdks/android/library/src/main/java/org/mozilla/fxaclient/internal/OAuthInfo.kt

```diff
@@ -6,34 +6,33 @@ package org.mozilla.fxaclient.internal
 
 import com.sun.jna.Pointer
 import com.sun.jna.Structure
-
-import java.util.Arrays
+import java.util.*
 
 class OAuthInfo internal constructor(raw: Raw) {
 
-    val accessToken: String?
-    val keys: String?
-    val scope: String?
+    val accessToken: String
+    val key: String?
+    val expiresAt: Date
 
     class Raw(p: Pointer) : Structure(p) {
         @JvmField var accessToken: Pointer? = null
-        @JvmField var keys: Pointer? = null
-        @JvmField var scope: Pointer? = null
+        @JvmField var key: Pointer? = null
+        @JvmField var expiresAt: Long = 0 // In seconds.
 
         init {
             read()
         }
 
         override fun getFieldOrder(): List<String> {
-            return Arrays.asList("accessToken", "keys", "scope")
+            return Arrays.asList("accessToken", "key", "expiresAt")
         }
     }
 
     init {
         try {
-            this.accessToken = raw.accessToken?.getRustString()
-            this.keys = raw.keys?.getRustString()
-            this.scope = raw.scope?.getRustString()
+            this.accessToken = raw.accessToken?.getRustString()!! // This field is always present.
```

Comment:
Should we try to keep this with a runtime exception handler? (like most of the code in a-c)

Reply:
I don't think it is worth it as serialization to AuthInfoC will fail if we can't create that string, but let's see what Thom says.

Reply:
Yes, this should never happen in practice, it's just hard to express given how JNA works.